Actions/WorkflowPostProcess/WorkflowPostProcess.ps1

```diff
@@ -1,26 +1,96 @@
 ﻿Param(
-    [Parameter(HelpMessage = "The event Id of the initiating workflow", Mandatory = $true)]
-    [string] $eventId,
     [Parameter(HelpMessage = "Telemetry scope generated during the workflow initialization", Mandatory = $false)]
-    [string] $telemetryScopeJson = '7b7d'
+    [string] $telemetryScopeJson = '',
+    [Parameter(HelpMessage = "The current job context", Mandatory = $false)]
+    [string] $currentJobContext = '',
+    [Parameter(HelpMessage = "The ref of the action", Mandatory = $false)]
+    [string] $actionsRef
 )
 
-$telemetryScope = $null
+function GetWorkflowConclusion($JobContext) {
+    # Check the conclusion for the current job
+    if ($JobContext -ne '') {
+        $jobContext = $JobContext | ConvertFrom-Json
+        if ($jobContext.status -eq 'failure') {
+            return "Failure"
+        }
+    }
 
-try {
-    . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
-    DownloadAndImportBcContainerHelper
-    import-module (Join-Path -path $PSScriptRoot -ChildPath "..\TelemetryHelper.psm1" -Resolve)
+    # Check the conclusion for the past jobs in the workflow
+    $workflowJobs = gh api /repos/$ENV:GITHUB_REPOSITORY/actions/runs/$ENV:GITHUB_RUN_ID/jobs --paginate -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" | ConvertFrom-Json
+    if ($null -ne $workflowJobs) {
+        $failedJobs = $workflowJobs.jobs | Where-Object { $_.conclusion -eq "failure" }
```

Comment:
From [the doc](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#list-jobs-for-a-workflow-run-attempt), possible values for `conclusion` are:

```
              "success",
              "failure",
              "neutral",
              "cancelled",
              "skipped",
              "timed_out",
              "action_required",
              null
```

Reply:
I've added some more checks for "cancelled" and "timed_out". Now, we only log "Success" if no jobs have state "failure", "cancelled" or "timed_out". 

Not sure we want to do anything for the remaining states. What do you think?

Reply:
No idea what `neutral` is. `action_required` shouldn't be ever the case. `skipped` would be interesting to see in some statistics.

If it's not too much trouble, maybe we can emit something like "5 jobs succeeded, 1 job skipped" or idk.

Reply:
Right now, I don't think we'll have any "Skipped" workflows but even if we did, I wonder what we/partners would do with that information in telemetry 🤷‍♂️ Doesn't seem super interesting IMO